### PR TITLE
THORN-2065 | Use smallrye-opentracing

### PR DESCRIPTION
**Pull request by pavolloffay**
Depends on: https://github.com/smallrye/smallrye-opentracing/issues/10

https://issues.jboss.org/browse/THORN-2065

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: testsuite/docker/opentracing-jaeger/src/it/java/io/thorntail/testsuite/opentracing/jaeger/OpenTracingJaegerAppIT.java [hunk 1]
```diff
@@ -64,42 +65,43 @@ public void test() throws Exception {
         System.err.println(tree);
         System.err.println("---");
 
-        assertThat(tree).hasRootSpans(2);
+        // TODO what is the first root span?
```
Comment (pavolloffay) [inline testsuite/docker/opentracing-jaeger/src/it/java/io/thorntail/testsuite/opentracing/jaeger/OpenTracingJaegerAppIT.java]: The other root span is, which is a bit weir. I think it's comming from jax-rs instrumentation and it should be linked to jax-rs span (maybe bug in `SpanNode` as it is not parsing `FOLLWOS_FROM` reference type).

> [ spanId: 938a897918d5ad5c, operation:deserialize, tags: {entity.type=java.lang.String, media.type=*/*}]

**Comment by pavolloffay** [comment]
This is ready to review/merge.

File: core/kernel/pom.xml [hunk 2]
```diff
@@ -46,19 +46,16 @@
     <dependency>
       <groupId>io.opentracing</groupId>
       <artifactId>opentracing-mock</artifactId>
-      <scope>provided</scope>
```
Comment (kenfinnigan) [inline core/kernel/pom.xml:49]: Why was "provided" removed from these dependencies?

@bobmcwhirter, I believe it's still needed right?

Or is "optional" enough?

Comment (bobmcwhirter) [reply core/kernel/pom.xml:49]: Optional may be sufficient.  If it's `optional`, I think that implies `provided`, but of course the inverse is not true.

**Review by bobmcwhirter — DISMISSED** [review]

**Comment by pavolloffay** [comment]
It says dismissed review bc I squashed commits

**Review by bobmcwhirter — APPROVED** [review]

**Pull request merged**